openfl/federated/task/fl_model.py

```diff
@@ -75,10 +76,17 @@ def __init__(self, build_model, optimizer=None, loss_fn=None, **kwargs):
             self.runner.validate = lambda *args, **kwargs: build_model.validate(
                 self.runner, *args, **kwargs
             )
+        # TODO: not used anywhere - should it be removed?
         if hasattr(self.model, "train_epoch"):
             self.runner.train_epoch = lambda *args, **kwargs: build_model.train_epoch(
                 self.runner, *args, **kwargs
             )
+
+        # Used to hook the training function when debugging locally
+        if hasattr(self.model, "train_"):
+            self.runner.train_ = lambda *args, **kwargs: build_model.train_(
+                self.runner, *args, **kwargs
+            )
```

Comment:
is it just for local debugging?

Reply:
When using a distributed deployment you can change the train function directly , so yes